srml/bridge/src/lib.rs

```diff
@@ -160,6 +161,39 @@ impl<T: Trait> Module<T> {
 	}
 }
 
+// A naive way to check whether a `child` header is an ancestor
+// of an `ancestor` header. For this it requires a proof which
+// is a header chain, This could be updated to use something like
+// Log2 Ancestors (#2053) in the future.
+fn verify_ancestry<H>(proof: Vec<H>, ancestor: H, child: H) -> std::result::Result<(), Error>
+where
+	H: Header
+{
+	let mut curr_header = &proof[0];
+	if curr_header.hash() != child.hash() {
+		return Err(Error::AncestorNotFound);
```

Comment:
I'd make the error `InvalidAncestryProof`. I think that reflects better that it is unknown whether one is an ancestor of the other, simply that we can't check because the proof is invalid.